### DISCUSSION TITLE
咨询下与EMQX的区别

### DISCUSSION THREAD

**Discussion by TxcA** [nanomq/nanomq#536] [Ideas]
主要目前是EMQX，无意在逛官方发现NanoMQ，感觉是边缘计算，搭建在终端的MQTT5.0 Broker吗?

**Comment by JaylinYu** [comment]
是的 NanoMQ是部署在边缘小设备的轻量级broker，区别在于EMQ在于：
1. 保持高并发高吞吐低时延的同时，大大减少了CPU和内存占用。但牺牲了分布式集群的支持。
2. 除了MQTT，未来还将支持ZMQ，nanomsg等常用的边缘总线协议。

**Comment by TxcA** [comment]
> 是的 NanoMQ是部署在边缘小设备的轻量级broker，区别在于EMQ在于：
> 
> 1. 保持高并发高吞吐低时延的同时，大大减少了CPU和内存占用。但牺牲了分布式集群的支持。
> 2. 除了MQTT，未来还将支持ZMQ，nanomsg等常用的边缘总线协议。

感谢，那是否考虑会出Android版本呢？(or java/kotlin)
场景需求:
1. 电视，可以直接作为IoT网关。
2. IoT设备，例如智能考勤机，做分布式考勤。
3. ...
目前自己所处行业，感觉基于Rk3288、Rk3399等方案的终端设备还是有一定保有量的。

**Comment by JaylinYu** [comment]
多谢您的建议，nanomq应可以直接通过交叉编译在android环境运行。后续会考虑退出官方的andorid版本。

**Comment by kami0314** [comment]
> > 是的 NanoMQ是部署在边缘小设备的轻量级broker，区别在于EMQ在于：
> > 
> > 1. 保持高并发高吞吐低时延的同时，大大减少了CPU和内存占用。但牺牲了分布式集群的支持。
> > 2. 除了MQTT，未来还将支持ZMQ，nanomsg等常用的边缘总线协议。
> 
> 感谢，那是否考虑会出Android版本呢？(or java/kotlin) 场景需求:
> 
> 1. 电视，可以直接作为IoT网关。
> 2. IoT设备，例如智能考勤机，做分布式考勤。
> 3. ...
>    目前自己所处行业，感觉基于Rk3288、Rk3399等方案的终端设备还是有一定保有量的。

赞同，强烈建议适配基于Rk3288、Rk3399的arm设备，OS: ubuntu/android

**Reply by JaylinYu** [reply]
Hi NanoMQ已经可以编译成so/.a 动态和静态库可以在APK里引用了

**Comment by JaylinYu** [comment]
Hi NanoMQ已经可以编译成so/.a 动态和静态库可以在APK里引用了

**Reply by TxcA** [reply]
抽时间研究下，感谢映云科技对这种小问题的跟进~
